components/compass-runtime-agent/cmd/main.go

```diff
@@ -102,8 +107,21 @@ func main() {
 	compassConnectionSupervisor, err := controllerDependencies.InitializeController()
 	exitOnError(err, "Failed to initialize Controller")
 
```

Comment:
This is just a test code for POC. Please notice that configProvider.GetRuntimeConfig() may just fail while compass-runtime-agent is waiting for configuration secret to be created. 

In the case of failing k3d jobs the configuration secret `compass-agent-configuration` from which you read runtime id is provided in the testing suite AFTER Kyma is fully installed on k3d. As a result  compass-runtime-agent should assume that this secret is missing and just wait until it is ready.